relay/stratoschain/grpc/query.go

```diff
@@ -100,3 +104,39 @@ func QueryMetaNode(p2pAddress string) (err error) {
 	}
 	return nil
 }
+
+func QueryTxByHash(txHash string) (err error) {
+	conn, err := CreateGrpcConn()
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+	client := sdktx.NewServiceClient(conn)
+	ctx := context.Background()
+	req := sdktx.GetTxRequest{Hash: txHash}
+	resp, err := client.GetTx(ctx, &req)
+	if err != nil {
+		return err
+	}
+
+	if resp == nil {
+		errMsg := fmt.Sprintf("QueryTxByHash returned nil response for transaction hash [%v]", txHash)
+		return errors.New(errMsg)
+	}
+	utils.Logf("--- resp is %v", *resp.TxResponse)
+	// skip non-successful tx
+	if resp.GetTxResponse().Code != 0 {
+		utils.ErrorLogf("Tx with hash[%v] failed: [%v]", txHash, resp.GetTxResponse().String())
+		return nil
+	}
+	// process relayed events
+	events := handlers.ProcessEvents(*resp.TxResponse)
+	for msgType, event := range events {
+		if handler, ok := handlers.Handlers[msgType]; ok {
+			go handler(event)
+		} else {
+			utils.ErrorLogf("No handler for event type [%v]", msgType)
+		}
+	}
```

Comment:
shouldn't be part of logic of this function

Reply:
solved